Clarify Rollback and Commit semantics

cipherboy opened the pull request:
Also fixes typo in read-only transaction definition.

DanGhita reviewed (approved):
Great, thank you Alex!

cipherboy commented:
> One more remark: I suppose we should have similar explanations in the comments from the new interfaces defined in "sdk/physical/transactions.go". Could you please check the comments are all aligned throughout the layers (logical, physical etc.).

Thanks, added. FWIW, only logical will be interacted with externally; physical is only accessible within a very narrow scope of Core, before barrier encryption is applied. Most everyone, even inside Core, will interact with barrier and thus the logical definitions. Updated!
